plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/navigation/openimplementation/OpenImplementationPresenter.java

```diff
@@ -96,161 +91,149 @@ public void show(final EditorPartPresenter editorPartPresenter) {
     final VirtualFile file = activeEditor.getEditorInput().getFile();
 
     if (file instanceof Resource) {
-      final Optional<Project> project = ((Resource) file).getRelatedProject();
+      final Project project = ((Resource) file).getProject();
 
       final Optional<Resource> srcFolder =
           ((Resource) file).getParentWithMarker(SourceFolderMarker.ID);
 
-      if (!srcFolder.isPresent()) {
+      if (project == null || !srcFolder.isPresent()) {
         return;
       }
 
       final String fqn = JavaUtil.resolveFQN((Container) srcFolder.get(), (Resource) file);
 
       service
-          .getImplementations(project.get().getLocation(), fqn, activeEditor.getCursorOffset())
+          .findImplementations(
+              new FindImplementationsCommandParametersDto(
+                  new FindImplementationsCommandParameters(
+                      project.getLocation().toString(), fqn, activeEditor.getCursorOffset())))
           .then(
-              new Operation<ImplementationsDescriptorDTO>() {
-                @Override
-                public void apply(ImplementationsDescriptorDTO impls) throws OperationException {
-                  int overridingSize = impls.getImplementations().size();
-
-                  String title =
-                      locale.openImplementationWindowTitle(impls.getMemberName(), overridingSize);
-                  NoImplementationWidget noImplementationWidget =
-                      new NoImplementationWidget(
-                          popupResources,
-                          javaResources,
-                          locale,
-                          OpenImplementationPresenter.this,
-                          title);
-                  if (overridingSize == 1) {
-                    actionPerformed(impls.getImplementations().get(0));
-                  } else if (overridingSize > 1) {
-                    openOneImplementation(
-                        impls, noImplementationWidget, (TextEditor) editorPartPresenter);
-                  } else if (!isNullOrEmpty(impls.getMemberName()) && overridingSize == 0) {
-                    showNoImplementations(noImplementationWidget, (TextEditor) editorPartPresenter);
-                  }
+              impls -> {
+                int overridingSize = impls.getImplementations().size();
+
+                String title =
+                    locale.openImplementationWindowTitle(impls.getMemberName(), overridingSize);
+                NoImplementationWidget noImplementationWidget =
+                    new NoImplementationWidget(
+                        popupResources,
+                        javaResources,
+                        locale,
+                        OpenImplementationPresenter.this,
+                        title);
+                if (overridingSize == 1) {
+                  actionPerformed(impls.getImplementations().get(0));
+                } else if (overridingSize > 1) {
+                  openOneImplementation(
```

Comment:
If we have more than one implementation we call a method that is call "openONEImplementation? Somethings seems misnamed here.